cpp/include/raft/linalg/detail/matmul.hpp

```diff
@@ -0,0 +1,293 @@
+/*
```

Comment:
In my opinion, this is a little different compared to the other wrapper code. In `detail/cublas_wrappers.hpp`, we have thin wrappers for unifying cuBLAS functions with different types and setting up the worker stream. For cuBLASLt `matmul`, we don't need it.
I think, the code in `detail/matmul.hpp` is more akin to `detail/gemm.hpp`, which introduces new functions that make use of the lower-level cuBLAS functions. It has different arguments and some logic to combine multiple cuBLASLt functions, caching, and some extra structures, etc. Hence I've put it in a dedicated `detail/matmul.hpp` to distance it a little bit from cuBLASLt.

That being said, I don't mind renaming `matmul.hpp` into `cublaslt_wrappers.hpp` and to host more cuBLASLt-related functions there later. Would you like me to proceed with the change?

Reply:
we aren't only using the wrappers for the types. We're using them to centralize and add a layer of indirection between RAFT and the cuda math libs. As an example, we shouldn't making any calls to the cublasLt libraries directly outside of those wrappers. There are, unfortunately, a couple remnants of legacy code in places where we're still calling cusparse directly, but that's the exception and not the rule here. This header file is doing the same thing as the other cuda math libs wrappers are doing (albeit a tad differently) and I'd like to keep the wrapper designs consistent so we aren't directly making any cublaslt calls outside of the wrapper itself. I'm even okay using raft::resources in the `cublastlt_wrappers.h` instead of passing the handle itself.

Reply:
Ok, got it! I moved the `matmul.hpp` to `cublaslt_wrappers.hpp`.